src/cobra/io/mat.py

```diff
@@ -179,6 +353,214 @@ def create_mat_metabolite_id(model: Model) -> str:
+    _d_replace: str
+        A string that points to the dictionary of converstions between MATLAB and
+        providers. Default D_MET (for metabolite).
+    """
+    annotation_matlab = list(
+        set(mat_struct.dtype.names).intersection(D_REPLACE[_d_replace].keys())
+    )
+    providers = [D_REPLACE[_d_replace][x] for x in annotation_matlab]
+    annotations = dict.fromkeys(providers, None)
+    _pumbed_re = re.compile("PMID: ?(\\d+),?")
+    _ec_re = re.compile(r"([\d\-]+.[\d\-]+.[\d\-]+.[\d-]+)")
+    _chebi_re = re.compile(r"\D*(\d+),?")
+    for name, mat_key in zip(providers, annotation_matlab):
+        if mat_key == "rxnReferences":
+            # This only picks up PMID: style references. Sometimes there are other
+            # things like PMC or OMIM, but those are ignored for now,
+            annotations[name] = _cell_to_str_list(
+                mat_struct[mat_key][0, 0], None, _pumbed_re
+            )
+        elif mat_key == "rxnECNumbers":
+            # turn EC codes to a list
+            annotations[name] = _cell_to_str_list(
+                mat_struct[mat_key][0, 0], None, _ec_re
+            )
+        elif mat_key == "metCHEBIID":
+            # if there are more than one CHEBI code, turn them to a comma separated str
+            annotations[name] = _cell_to_str_list(
+                mat_struct[mat_key][0, 0], None, _chebi_re
+            )
+        else:
+            # If it something else, which may have commas, turn it into a list
+            annotations[name] = [
+                [y.strip() for y in _each_cell.split(", ")] if _each_cell else None
+                for _each_cell in _cell_to_str_list(mat_struct[mat_key][0, 0])
+            ]
+    for i, obj in enumerate(target_list):
+        obj.annotation = {
+            prov: annotations[prov][i] for prov in providers if annotations[prov][i]
+        }
+
+
+def mat_parse_notes(
+    target_list: List[Object],
+    mat_struct: np.ndarray,
+    _d_replace: str = D_REACTION_NOTES,
+) -> None:
+    """Process mat structure notes in place.
+
+    Will process mat structured notes and add them to a list of new entities
+    (metabolites, reactions, genes) in a format based on identifiers.org.
+
+    Parameters
+    ----------
+    target_list: list[cobra.Object]
+        A list of cobra objects, including metabolites, reactions or genes. The
+        notes will be added to these lists.
+    mat_struct: np.ndarray
+        A darray that includes the data imported from matlab file.
+    _d_replace: str
+        A string that points to the dictionary of converstions between MATLAB and
+        notes. Default D_REACTION_NOTES (for reactions).
+    """
+    annotation_matlab = list(
+        set(mat_struct.dtype.names).intersection(D_REPLACE[_d_replace].keys())
+    )
+    note_providers = [D_REPLACE[_d_replace][x] for x in annotation_matlab]
+    notes = dict.fromkeys(note_providers, None)
+    _pumbed_re = re.compile("PMID: ?(\\d+),?")
```

Comment:
Typo corrected. I feel better keeping it starting with _, since there are multiple variants of it. Each one is limited in scope, but that tells me it should be limited when reading the code.